File: calculator.php

<?php
  if ($_SERVER["REQUEST_METHOD"] == "POST") {
    $num1 = $_POST["num1"];
    $num2 = $_POST["num2"];
    $operator = $_POST["operator"];

    if (empty($num1) || empty($num2)) {
      $error = "Please enter both numbers.";
    } else if (!is_numeric($num1) || !is_numeric($num2)) {
      $error = "Invalid input. Please enter numbers only.";
    } else {
      $result = "";
      switch ($operator) {
        case "+":
          $result = $num1 + $num2;
          break;
        case "-":
          $result = $num1 - $num2;
          break;
        case "*":
          $result = $num1 * $num2;
          break;
        case "/":
          if ($num2 == 0) {
            $error = "Cannot divide by zero.";
          } else {
            $result = $num1 / $num2;
          }
          break;
      }
    }
  }
?>

<form method="post">
  <?php if (isset($error)) : ?>
    <p style="color: red;"><?php echo $error; ?></p>
  <?php endif; ?>
  <input type="number" name="num1" placeholder="Enter first number">
  <select name="operator">
    <option value="+">+</option>
    <option value="-">-</option>
    <option value="*">*</option>
    <option value="/">/</option>
  </select>
  <input type="number" name="num2" placeholder="Enter second number">
  <input type="submit" value="Calculate">
</form>

<?php if (isset($result)) : ?>
  <p>The result is: <?php echo $result; ?></p>
<?php endif; ?>

File: Calculator.py
def calculate(num1, num2, operator):
  """Performs the specified arithmetic operation on two numbers.

  Args:
      num1: The first number.
      num2: The second number.
      operator: The arithmetic operator (+, -, *, or /).

  Returns:
      The result of the operation.
  """
  if operator == "+":
    return num1 + num2
  elif operator == "-":
    return num1 - num2
  elif operator == "*":
    return num1 * num2
  elif operator == "/":
    if num2 == 0:
      print("Error: Cannot divide by zero.")
      return None
    else:
      return num1 / num2
  else:
    print("Invalid operator.")
    return None

while True:
  # Get user input
  num1 = float(input("Enter the first number: "))
  num2 = float(input("Enter the second number: "))
  operator = input("Choose an operator (+, -, *, /): ")

  # Perform calculation
  result = calculate(num1, num2, operator)

  # Display result
  if result is not None:
    print(f"The result is: {result}")

  # Ask if user wants to continue
  choice = input("Do you want to perform another calculation? (y/n): ")
  if choice.lower() != "y":
    break

print("Calculator closed.")
